### PR TITLE
Added customizable scopes

### PR DESCRIPTION
**Pull request by itsMeDavidV**
Edited TeslaAPI enum to include customizable scopes for Fleet API. Did not touch .ownerAPI as Owner API is deprecated and its deletion from this package should occur soon.

### CODE REVIEW

**Comment by jonasman** [comment]
Can you add a default Scope to the enum to return all the standard scopes so that we dont need to think too much when adding scopes?

File: Sources/TeslaSwift/TeslaSwift.swift [hunk 1]
```diff
@@ -29,14 +29,25 @@ public enum TeslaAPI {
...
+    
+    public enum Scope: String {
+        case openId = "openid"
+        case offlineAccess = "offline_access"
+        case userData = "user_data"
+        case vehicleDeviceData = "vehicle_device_data"
+        case vehicleCmds = "vehicle_cmds"
+        case vehicleChargingCmds = "vehicle_charging_cmds"
+        case energyDeviceData = "energy_device_data"
+        case energyCmds = "energy_cmds"
```
Comment (jonasman) [inline Sources/TeslaSwift/TeslaSwift.swift:41]: we can make a var here that returns the common vehicle scopes

File: Sources/TeslaSwift/TeslaSwift.swift [hunk 2]
```diff
@@ -29,14 +29,25 @@ public enum TeslaAPI {
...
+    }
 
     case ownerAPI
-    case fleetAPI(region: Region, clientID: String, clientSecret: String, redirectURI: String)
+    case fleetAPI(region: Region, clientID: String, clientSecret: String, redirectURI: String, scopes: [Scope])
```
Comment (jonasman) [inline Sources/TeslaSwift/TeslaSwift.swift:45]: here would could do  scopes: `[Scope] = .default`

**Comment by itsMeDavidV** [comment]
> Can you add a default Scope to the enum to return all the standard scopes so that we dont need to think too much when adding scopes?

Sure, standard scopes as in just vehicle-related + openid + offline or all?

**Comment by jonasman** [comment]
@itsMeDavidV like adding all to the variable in the enum

**Comment by jonasman** [comment]
Hi, i added myself the `static var all`  so we dont need to set the scopes manually

**Comment by itsMeDavidV** [comment]
Nice, sorry about that! Not a lot of spare time these days :/

**Pull request merged**